classes/local/store/object_file_system.php

```diff
@@ -737,7 +737,26 @@ protected function copy_from_external_to_local($contenthash) {
      */
     public function redirect_to_presigned_url($contenthash, $headers = array()) {
         try {
-            redirect($this->externalclient->generate_presigned_url($contenthash, $headers));
+            $signedurl = $this->externalclient->generate_presigned_url($contenthash, $headers);
+            if (headers_sent()) {
+                debugging('Headers already sent! This redirect may be incorrectly cached in browser');
+            } else {
+                // Remove all previously-set headers.
+                foreach (headers_list() as $header) {
+                    // Get header name (text before the colon).
+                    if (preg_match('~^([^:]+):~', $header, $matches)) {
+                        header_remove($matches[1]);
+                    }
+                }
```

Comment:
This might be more simply done as such:
```
 header_remove(null);
```
Ref: https://www.php.net/manual/en/function.header-remove.php

Reply:
See Sam's previous comment